### ISSUE TITLE
Replace the console driven main menu with swing UI-based main menu

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#356]
We have been relying upon the [console](https://github.com/mars-sim/mars-sim/wiki/Mars-Net-Console) terminal for configuring the mars-sim before starting the simulation.

No doubt, in Headless Edition, we will continue to make use of the console as the major method for interaction. 

In any cases, setting up [Remote Connection]( https://github.com/mars-sim/mars-sim/wiki/Remote-Console-Connection) via SSH client is available as an option, which will pave the way for future networking capability of mars-sim.

Seems it's about time we rebuild the Main Menu using Swing UI for the Swing Edition to make it feel more complete.

Q: are there any packages (that have maven artifacts) out there that will help facilitate our job ?

Also, we should add back displaying a splash screen during the startup.

> Note : the caveat is that the console does use swing components to build because the [beryx / text-io](https://github.com/beryx/text-io) package that we depend on is using [JLine](https://github.com/jline/jline2).

**Comment by bevans2000** [comment]
I like the terminal interface, gives a quick way to examine and change things. I will have a look when I get back home

**Comment by bevans2000** [comment]
What do you mean by this? You want to see a startup menu that allows the user to select what to start? The command line args do drive it but I guess you are talking about the novice user

**Comment by mokun** [comment]
If the player doesn't bypass the startup, it will go into the Main Menu in the console window.

```
 ---------------  M A R S   S I M U L A T I O N   P R O J E C T  ---------------

                          * * *   Main Menu   * * *


0. Exit
1. Start a new Sim
2. Load a saved Sim
3. Change screen resolution

Choose an option : ​
```

This main menu is in text and not made in swing UI for players.

**Comment by mokun** [comment]
See the logs : 

```
2025-09-20T16:00:03.5 (Config) MarsProject : Starting Mars Simulation Project pre-3.10.0 - 64-bit Java 21.0.7 - 10 Cores
2025-09-20T16:00:03.5 (Info) SimulationConfig : Loading simulation configuration from /Users/spacebear/.mars-sim/xml
2025-09-20T16:00:03.9 (Config) MarsProject : Do you want to bypass the console menu and start a new default simulation in Sandbox Mode ?
2025-09-20T16:00:03.9 (Config) MarsProject : To proceed, please choose 'Yes' or 'No' button in the dialog.
2025-09-20T16:00:06.4 (Config) MarsProject : Please go to the Console Main Menu to choose an option.
2025-09-20T16:00:06.4 (Config) InteractiveTerm : Staring startConsoleMainMenu()

```

**Comment by mokun** [comment]
Choosing 1, 2, 1 will get the player to start the Scenario Editor.

```
 ---------------  M A R S   S I M U L A T I O N   P R O J E C T  ---------------

                          * * *   Main Menu   * * *


0. Exit
1. Start a new Sim
2. Load a saved Sim
3. Change screen resolution

Choose an option : ​1

Starting a new sim...

 ---------------  M A R S   S I M U L A T I O N   P R O J E C T  ---------------

                        * * *   Mode Selection   * * *


0. Exit
1. Command Mode (Experimental only)
2. Sandbox Mode
3. Society Mode (Not Available)
4. Sponsor Mode (Experimental only)

Select the Game Mode: ​2

Go to Sandbox Mode.


           * * *  Sandbox Mode - Crew and Scenario Selection  * * *

0. Exit
1. Open Scenario Editor
2. Enable/disable crew loading (currently Enabled)
3. Proceed to start the sim
4. Back to the previous menu

Note 1: By default the loading of crews is Enabled. Choose 2 to enable/disable.
Note 2: The Console Editon does not come with the Scenario Editor.

Enter your choice: ​1

Loading the Scenario Editor in Swing-based UI...

```

This combination of both console text and Swing UI can be confusing.

Ideally, it should be either purely console text based or swing UI based. 

Now it's a bit of both.

**Comment by mokun** [comment]
> what about GamesMode? I've never understood it.

As you know, right now, we've only focused on developing `Sandbox` mode.

We used to develop the `Command` mode game play, which simply means that a player only has one settlement to be concerned with. 

All other settlements are AI's and this player is supposed to barred from seeing the intricate details of other settlements. 

But if he trades and form pacts with them, through some agreement they may reveal more of their stats to him.

A `Sponsor` mode would mean the player is playing an authority and may have multiple settlement to start with.

That's how I see the kind of challenge a player is to be presented with if he plays the `Command` or `Sponsor` mode.

**Comment by bevans2000** [comment]
Hmm, I think I'll do the initial main menu for Sandbox mode as a starter. Probably the Commander mode needs revisiting everywhere because the all the recent changes do not cater for Commander mode. 
I wonder if instead of having a dedicated window we just restrict the Unit windows to be read only or not.

**Comment by bevans2000** [comment]
I've got a much simpler start-up working with less waiting and statics. Much cleaner and lower coupling.
There is one weird problem with the UI doesn't start updating for a couple of seconds at the initial appearance. Sure I will get to the bottom of it.

**Comment by mokun** [comment]
Awesome !

> There is one weird problem with the UI doesn't start updating for a couple of seconds at the initial appearance.

Currently in Eclipse IDE, right after I start the simulation, from time to time (say 10% chance) the time would be at 000 milisols and won't move forward.

If that happens, I would click on pause speed button and hit it again to play. This fixes it.

I don't know why this happens.

Do you think what you encounter is related to this or they are separate thing ?

**Comment by bevans2000** [comment]
> Currently in Eclipse IDE, right after I start the simulation, from time to time (say 10% chance) the time would be at 000 milisols and won't move forward.
> 
> If that happens, I would click on pause speed button and hit it again to play. This fixes it.
> 
> I don't know why this happens.
> 

I found this problem, it relates to the recent Master clock changes you did. Sometimes the first pulse has a leadPulseTime of zero which mean a pulse is never generated and hence it goes into a loop of not doing anything. Pausing and restarting forces a recalculation.
The changes make no sense to me and look very complex. For now I have added a patch to force the leadPulseTime to a small value if it is zero; this will force it out of the useless loop.

**Comment by mokun** [comment]
> For now I have added a patch to force the leadPulseTime to a small value if it is zero; 

Can you commit the change first so that I can have it ?

**Comment by bevans2000** [comment]
@mokun tell me what you thing of the start chooser? It might need a bit more decoration like an image but functionaly it works well. It's a good addition.

**Comment by mokun** [comment]
> tell me what you thing of the start chooser?

what *switch* do I need to add to my eclipse launcher in order to load the start chooser ?

**Comment by bevans2000** [comment]
The chooser only runs if there are no arguments. If there is a scenario or template on the command line it should not be used. It pops up on the no args case

**Comment by mokun** [comment]
> The chooser only runs if there are no arguments.

oic.

Yes it's great to have a quick and simple startup chooser instead of going back and forth with console and swing dialog.

**Comment by mokun** [comment]
Now, if I click on the first button `New Simulation`, it's no different from clicking on *From a Settlement Template` and choosing *Default* to start a default scenario.

Or should we create something different when choosing `New Simulation` ?

**Comment by bevans2000** [comment]
Yes that makes sense. The New Simulation just mirrors what we had before; it's a quick shortcut to a new simulation using the default Scenario. So selecting "From Scenario" and choosing the default is the same. But if you had more Scenarios, like me, you would have more choice available,
Try the Custom Scenario option and create a new Scenario.
To be clear, the chooser just reflects the existing command line logic but is easier for the user to access. I think this is a good step forward.
I want to also create a Windows & Linux installer that bundles the JVM using jpackage so it becomes easier to run.
